website/content/en/status/report-2023-04-2023-06/kboot.adoc

```diff
@@ -0,0 +1,55 @@
+=== LinuxBoot FreeBSD on aarch64
+
+Contact: Warner Losh <imp@bsdimp.com>
+
+Links: +
+link:https://www.linuxboot.org/[LinuxBoot Project] URL: link:https://www.linuxboot.org/[]+
+link:https://docs.google.com/presentation/d/1N5Jp6XzYWv9Z9RhhETC-e6tFkqRHvp-ldRDW_9h2JCw/edit?usp=sharing[BSDCan 2023 kboot talk slides] URL: link:https://docs.google.com/presentation/d/1N5Jp6XzYWv9Z9RhhETC-e6tFkqRHvp-ldRDW_9h2JCw/edit?usp=sharing[]
+
+LinuxBoot is an effort to create a clean, robust, auditable and repeatable boot firmware.
+What originally started as a specific project at Google has grown to encompass any boot environment that uses Linux to launch the final operating system.
+Many platfors now support this environment, in some cases it is the only available boot environment.
+In addition, some embedded boxes have a LinuxBoot environment hard-coded that is quite hard to change, and being able to reboot into FreeBSD is desirable.
+
+The old Sony PlayStation 3 port used a boot loader called 'kboot' to boot the FreeBSD port from its Linux kernel (all predating the LinuxBoot project).
+That code has been greatly expanded, made generic with easily replaceable per-architecture plug ins.
+The normal FreeBSD [.filename]#/boot/loader# is built as a Linux binary that reads in the FreeBSD kernel, modules and tunables.
+It places them into memory as if it were running in a pre-boot environment, then loads that image into the Linux kernel with man:kexec_load[2] and does a special reboot to that image.
```

Comment:
```suggestion

The normal FreeBSD [.filename]#/boot/loader# is built as a Linux binary that reads in the FreeBSD kernel, modules, and tunables.

LinuxBoot places the FreeBSD kernel, modules, and tunables into memory as if it were running in a pre-boot environment, then loads that image into the Linux kernel with man:kexec_load[2] and performs a special reboot to the image.
```

Disambiguate the words 'It' and 'them' … correct, or have I misunderstood?

(I might be slaving over this unnecessarily; the report acknowledges the need for a manual page.)

Reply:
as if loader.kboot were running as loader.efi and trying to load the kernel, tuneables and modules from there (similar memory layouts happen). I struggled with a way to say loader.kboot will present the same environment to the kernel's startup code after loader.kboot does its kexec that loader.efi does after loading everything and jumping to the kernel's startup code.

Reply:
> … I struggled with a way to say …

How about … a freehand sketch? (As if you have nothing better to do.) 

It needn't be perfect; the absence of loader.kboot(8) is uppermost amongst your pleas for help.

File an image somewhere under `/home/imp/public_html` then it can be linked from within the report. Just a thought.

Reply:
If an image is to be included, please remember that the report will also be sent by email. So either we include a `link` reference or maybe a picture in Ascii art.

But a good wording is probably better.